compiler/quilt/tools/data_transfer.py

```diff
@@ -152,14 +152,23 @@ def _worker_thread():
                                 encoding = response.raw.headers.pop('Content-Encoding', None)
 
                                 # Make sure we're getting the expected range.
-                                content_range = response.headers.get('Content-Range', '')
-                                match = CONTENT_RANGE_RE.match(content_range)
-                                if not match or not int(match.group(1)) == range_start:
-                                    with lock:
-                                        tqdm.write("Unexpected Content-Range: %s" % content_range)
-                                    break
+                                if response.status_code == 200:  # server ignored range request
+                                    compressed_size = response.headers.get('Content-Length', '').strip()
+                                    if not compressed_size:
+                                        # HTTP: Content length is specified by 'Content-Length' header, or
+                                        # by closing the connection when done. :-/
+                                        compressed_size = len(response.content)
```

Comment:
This forces the whole contents to be downloaded - and loaded into memory - right? So probably not a good idea. Does it actually happen in practice? If not, I'd just fail in this case.

Reply:
Then we can give up on Google Cloud Platform compatibility.

My *suspicion* is that GCP Storage ignores the range request if the file is too small to warrant it, since their stated documentation is that they support the range request.  However, they do not always return a 206 response (and aren't really required to by the HTTP standard, as it's an optional server feature).

We could, however print a warning (via `tqdm`), like "Range request ignored by S3 server, file will be fully loaded into memory: <s3 path>", perhaps.

Reply:
Well, it's fine if it ignores the range request - but does it really not return `Content-Length`?

Reply:
It's not that Google doesn't send `Content-Length`, it's that no server is required to for a regular 200 response.  I mean, they should, who wouldn't, right?  But http doesn't require it.